crates/ln-dlc-node/src/ldk_node_wallet.rs

```diff
@@ -221,20 +226,16 @@ where
     #[autometrics]
     pub(crate) fn get_last_unused_address(&self) -> Result<bitcoin::Address, Error> {
         let address_info = tokio::task::block_in_place(move || {
-            self.runtime_handle.block_on(async move {
-                self.inner
-                    .lock()
-                    .await
-                    .get_address(AddressIndex::LastUnused)
-            })
+            self.runtime_handle
+                .block_on(async move { self.bdk_lock().get_address(AddressIndex::LastUnused) })
```

Comment:
🤔 there is no await here anymore, is the async still needed?

This also applies to the other methods below.

- get_balance
- on_chain_transaction_list

Reply:
it might be not needed! maybe we can remove the `runtime_handle` as a result!

Reply:
`runtime_handle` needs to stay for now for the fee estimation work, so I'll not touch it especially that @luckysori has a PR that refactors it completely.

I'll change `on_chain_transaction_list` in a separate follow-up, because there's a lot of unnecessary async calls in the code that will drop away and I don't want to cloud this one.